linux/power_operation_scripts/set_power_script.yml

```diff
@@ -78,33 +82,64 @@
 - name: "Update inventory"
   include_tasks: ../../common/update_inventory.yml
 
-- name: "Check the script file running as expected"
-  ansible.builtin.stat:
-    path: "{{ script_tag_path }}"
+- name: "Wait for VMware Tools running"
+  include_tasks: ../../common/vm_wait_vmtools_status.yml
+  vars:
+    vm_wait_vmtools_running: true
+
+- name: "Get content of file tools.conf after {{ power_script_type }} {{ power_script_op }}"
+  ansible.builtin.command: "cat {{ vmtools_config_file }}"
+  register: tools_conf_content_result
   delegate_to: "{{ vm_guest_ip }}"
-  register: tag_path_exist_result
-  failed_when: tag_path_exist_result.stat.exists is undefined or not tag_path_exist_result.stat.exists
+
+- name: "Check the script file running as expected"
   when:
     - script_tag_path is defined and script_tag_path
     - power_script_type == "custom"
     - power_script_op != "suspend"
+  block:
+    - name: "Get the status of file {{ script_tag_path }} for {{ power_script_type }} {{ power_script_op }}"
+      ansible.builtin.stat:
+        path: "{{ script_tag_path }}"
+      delegate_to: "{{ vm_guest_ip }}"
+      register: tag_path_exist_result
+      until:
+        - tag_path_exist_result.stat.exists is defined
+        - tag_path_exist_result.stat.exists
+      retries: 30
+      delay: 5
+
+    - name: "Check the serching result"
+      ansible.builtin.assert:
+        that:
+          - tag_path_exist_result.stat.exists is defined
+          - tag_path_exist_result.stat.exists
+        fail_msg: "Failed to find {{ script_tag_path }} for {{ power_script_type }} {{ power_script_op }}."
+        success_msg: "Successfully found the {{ script_tag_path }} for {{ power_script_type }} {{ power_script_op }}."
 
 - name: "Check the log file"
   when: power_script_op != "suspend"
   block:
-    - name: "Search the log file"
-      ansible.builtin.shell: "grep '{{ script_file_path }}' /var/log/vmware-tools/*.log"
+    - name: "Search the log file {{ script_file_path }}"
+      ansible.builtin.shell: "grep {{ script_file_path | basename }} *.log"
```

Comment:
why do you replace file path with basename at here?

Reply:
change back to search file path.